examples/kotlin/src/main/kotlin/com/example/booktest/postgresql/QueriesImpl.kt

```diff
@@ -92,43 +88,35 @@ WHERE book_id = ?
 class QueriesImpl(private val conn: Connection) : Queries {
 
   @Throws(SQLException::class)
-  override fun booksByTags(dollar1: List<String>): ListQuery<BooksByTagsRow> {
-    return object : ListQuery<BooksByTagsRow>() {
-      override fun execute(): List<BooksByTagsRow> {
-        return conn.prepareStatement(booksByTags).use { stmt ->
-          this.statement = stmt
-          stmt.setArray(1, conn.createArrayOf("pg_catalog.varchar", dollar1.toTypedArray()))
-
-          val results = stmt.executeQuery()
-          val ret = mutableListOf<BooksByTagsRow>()
-          while (results.next()) {
-              ret.add(BooksByTagsRow(
+  override fun booksByTags(dollar1: List<String>): List<BooksByTagsRow> {
+    return conn.prepareStatement(booksByTags).use { stmt ->
+      stmt.setArray(1, conn.createArrayOf("pg_catalog.varchar", dollar1.toTypedArray()))
+
+      val results = stmt.executeQuery()
+      val ret = mutableListOf<BooksByTagsRow>()
+      while (results.next()) {
+          ret.add(BooksByTagsRow(
```

Comment:
I don't know how JDBC implementations are implemented these days but the JDBC API has been built to encourage "streaming" where not the entire result set has to be sent over the wire before it can start being processed by the application. This implementation seems to break that. I think you can build a list implementation on top of ResultSet using AbstractSequentialList quite easily? You would have to convert the checked `SQLException` to unchecked exceptions though so that might be a problem...

Reply:
Actually, doesn't Kotlin not have checked exceptions?

Reply:
The "streaming" method is supported in Go too. It's a lower level usage that sqlc doesn't currently handle because it wants to abstract away row parsing. I think if @kyleconroy adds a command for the Go version, like `:execstream`, we can replicate it for Kotlin. Otherwise this should match the Go impl to always parse the results first.